Arte marcial para el HoS, "Seclat"

SagaStars opened the pull request:
_intente sacar gifs del los action buttons del seclat pero mi pc sucks_

**NUEVO ARTE MARCIAL, "SECLAT"**
Es un arte marcial conformado de 6 mayores características , diseñado para las cabezas de seguridad para que se les haga una tarea mas "facil" combatir contra antags/payasos a corto alcance, con estilo, MUCHO ESTILO.

### 1.- Heart Break Shot
![h](https://user-images.githubusercontent.com/55407528/69498435-72c04600-0ebe-11ea-81c0-b590ff84d6f0.png)
Es un golpe directo al corazón que infligirá un knockback de 1 tile, 13 de daño `BRUTE` en el pecho, 14 de daño de `STAMINA`, te dara un poco de `loosebreath` y te dara un ligero stun en el piso por menos de un segundo, es una referencia a hajime no ippo muy fuerte y lo elegí por lo estúpidamente estrafalario que es y lo genial que le vendria al HoS _(De igual forma ni tan estupido es, ya tenemos un golpe que te da directamente en el pulmón, lmao)_. En conclusiòn es como el **Leg Sweep** y **Lung Punch** del Warden fusionados en una versión reducida de ambos.

### 2.- SUPLEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEX
Cuando desarmas a alguien que tienes agarrado efectuas un **PUTOOO SUPLEEEEEX**, Inflige un 36 de daño `BRUTE` efectuando la animacion del *flip a la vez de tirando a los dos al suelo por 2 segundos.

### 3.- Advanced Neck Chop
![h](https://user-images.githubusercontent.com/55407528/69498639-6b9a3780-0ec0-11ea-93f6-1c17d6f2b96c.png)
No mucho que decir, es el **Neck Chop** del Warden pero por mas tiempo y con un poco mas de daño. Muy útil para la gente que tiene bombas que se activan por voz. (Lo digo de experiencia.)

### 4.- Probabilidad de Bloqueo
El HoS tendrá un 35% de bloqueo de ataques físicos ÚNICAMENTE cuando este tenga un objeto en la mano y este en throw mode (Debido a eso la gran probabilidad).

### 5.- Cambios a los golpes básicos 
Tendrá un cambio en los ataques a mano desnuda, ya que ahora el flavor text va a intervenir un poco en el daño al oponente. Habran 4 tipos de ataque diferente cada uno con sus respectivos bonus de daño.
Daño de una **Inverse Kick**: `14 BRUTE` 25% Prob. Del Bonus
Daño de una **Backflip Kick**: `16 BRUTE + Una animación de flip` 25%  Prob. Del Bonus
Daño de un **Pivot Jab** y **Throws a flurry of punches at**: `12 BRUTE` 50% Prob. Del Bonus
Daño a un oponente en el suelo, aparece como "**Beats the shit out of**": `16 BRUTE` 100% Prob. Del Bonus

### 6.- Agarre Agresivo
Cuando efectúas un agarre instantáneamente es agresivo, por lo cual es mas difícil de zafarse del agarre, también puedes tirar a tu enemigo rápidamente.

### BONUSSSS.- El HoS ya no puede usar el Krav Maga.
Pues eso, cuando el hos se equipa los guantes no tienen efecto alguno a la par de un texto que te lo indica.

## Changelog
:cl: 
add: Un nuevo arte marcial "seclat" con el cual spawneara junto al hos
tweak: El HoS no puede usar las habilidades del Krav Maga (Tampoco el 60% de Prob. de desarme)
:cl:

SagaStars committed:
marte marcial pal hos

Evankhell561 commented on code/HISPANIA/modules/martial_arts/seclat.dm:
```diff
@@ -0,0 +1,128 @@
+			streak = ""
+			advanced_neck_chop(A,D)
+			return 1
+	return 0
```
Noto que tienes muchos return 1 y return 0. Vas a infartar a Ryzor, te recomiendo cambiar los return 1 por return TRUE y los return 0 por return FALSE.

SagaStars replied:
Uy... bien bien ya lo cambio, y si me permites preguntar cual es la diferencia?

Evankhell561 replied:
Ninguna, Ryzor te dirá para que sea más claro. En tu codigo los 1 están para decir que algo es verdadero y los 0 para decir que es falso, así que es mejor usar TRUE Y FALSE que para eso están, Los numeros deberian ser usados para hacer operaciones matáticas, no para representar falso o verdadero.

Evankhell561 commented on code/modules/martial_arts/krav_maga.dm:
```diff
@@ -125,7 +132,9 @@ datum/martial_art/krav_maga/grab_act(var/mob/living/carbon/human/A, var/mob/livi
-/datum/martial_art/krav_maga/disarm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+/datum/martial_art/krav_maga/disarm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D, var/mob/living/carbon/human/H)
+	if(H.mind.assigned_role == "Head of Security") // De este modo el hos no adquiere el 60% de desarme del krav maga.
+		return
```
En este caso deberias poner return FALSE (o return 0)

SagaStars replied:
Yea got it

SagaStars committed:
ryzor ya no me puede castrar B)

- cambio los `return 1` por `return true` y los `return 0` por `return false`

SagaStars committed:
Whoops, se me ha olvidado el proc de ayuda

- Ahora hay un `remember_the_basics` para el seclat.
- Subida de 30 a 35 en `block_chance`.

DanaDririon commented on code/modules/martial_arts/krav_maga.dm:
```diff
@@ -105,7 +110,9 @@ datum/martial_art/krav_maga/grab_act(var/mob/living/carbon/human/A, var/mob/livi
 	..()
 
-/datum/martial_art/krav_maga/harm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+/datum/martial_art/krav_maga/harm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D, var/mob/living/carbon/human/H)
```
Creo que la variable H que agregaste está de más
Puedes usar el A que ya está

SagaStars replied:
true, ya en breve lo arreglo

SagaStars replied:
**E N  B R E V E**

DanaDririon commented on code/HISPANIA/modules/martial_arts/seclat.dm:
```diff
@@ -0,0 +1,141 @@
+	set category = "Seclat"
+	to_chat(usr, "<b><i>You try to remember some of the basics of Seclat.</i></b>")
+
+	to_chat(usr, "<span class='notice'>Heart Break Shot</span>: A powerful corkscrew to the chest, stopping the heart for less than a second. Deals stamina damage and lauches the enemy forward.")
```
launches

SagaStars replied:
el buen castellano yo soy

DanaDririon commented on code/HISPANIA/modules/martial_arts/seclat.dm:
```diff
@@ -0,0 +1,141 @@
+	to_chat(usr, "<span class='notice'>Suplex</span>: Disarm someone you are grabbing. Suplexes your target to the floor. Greatly injures them and leaves both you and your target on the floor.")
+	to_chat(usr, "<span class='notice'>Advanced Neck Chop</span>: Injures the neck, stopping the victim from speaking for a while. Deals an average amount of damage.")
+
+	to_chat(usr, "<b><i>In addition</span>: Having your throw mode on when being attacked, you enter an active defense mode where you have a chance to block. Also all of your grabs will be aggresive and your standar attacks will be more powerful as well.</i></b>")
```
standard

SagaStars replied:
orthographia

SagaStars committed:
arregls 0rthograpicos y muerte a H

iRyzor commented:
Este PR debe ser testeado por la comunidad de devs porque tiene pinta de que el HoS puede destruirte la vida con la mirada.

SagaStars commented:
> Este PR debe ser testeado por la comunidad de devs porque tiene pinta de que el HoS puede destruirte la vida con la mirada.

Understandable

SagaStars committed:
Merge branch 'master' into seclat

ghost commented:
Lmao que op, lo del suplex tiene pinta de molar.

SagaStars committed:
Merge branch 'master' into seclat

SagaStars committed:
Merge branch 'master' into seclat

SagaStars commented:
EVAAAAN, EL PR ESTA AGARRANDO MUCHO POLVO

SagaStars committed:
Merge branch 'master' into seclat

SagaStars commented:
El rework del seclat vendra en un PR aparte.
